assembl/graphql/schema.py

```diff
@@ -1,81 +1,439 @@
+        return resolve_langstring(self.description, args.get('lang'))
+
+    def resolve_questions(self, args, context, info):
+        return self.get_children()
+
+    def resolve_video(self, args, context, info):
+        title = resolve_langstring(self.video_title, args.get('lang'))
+        description = resolve_langstring(self.video_description, args.get('lang'))
+        return Video(
+            title=title,
+            description=description,
+            html_code=self.video_html_code,
+        )
+
+    def resolve_img_url(self, args, context, info):
+        # TODO imgUrl
+        return ""
 
 
 class Query(graphene.ObjectType):
-    node = URINode.Field()
-    agent_profiles = SQLAlchemyConnectionField(AgentProfile)
-    agent_profile = graphene.Field(AgentProfile)
-    user = graphene.Field(User)
-
-
-Schema = graphene.Schema(query=Query)
-
-"""
-query {
-    getUserAgent(id: $ID){
-        edge {
-            whatever
-        }
-    }
-}
-"""
-
-# this can execute:
-# schema.execute("query {node(id:\"local:AgentProfile/3\") {id ... on AgentProfile {name}} }")
-# schema.execute("query {agentProfiles {edges { node {id name}}} }")
-# This fails:
-# schema.execute("query {agentProfiles {edges { node {id name   ... on User {preferredEmail}}}} }")
-# schema.execute("query {agentProfiles(id:\"local:AgentProfile/3\") {id name} }")
+    node = Node.Field()
+    posts = SQLAlchemyConnectionField(PostConnection)#, idea_id=graphene.ID())
+    # ideas = SQLAlchemyConnectionField(Idea)
+    thematics = graphene.List(Thematic, identifier=graphene.String())
+    # agent_profiles = SQLAlchemyConnectionField(AgentProfile)
+
+    def resolve_ideas(self, args, context, info):
+        # TODO do we need to do a special query to only retrieve ideas
+        # associated to the RootIdea?
+        connection_type = info.return_type.graphene_type  # this is IdeaConnection
+        model = connection_type._meta.node._meta.model  # this is models.Idea
+        query = get_query(model, context)
+        discussion_id = context.matchdict['discussion_id']
+        query = query.filter(model.discussion_id == discussion_id
+            ).filter(model.hidden == False
+            ).filter(model.tombstone_condition())
+
+        # pagination is done after that, no need to do it ourself
+        return query
+
+    def resolve_posts(self, args, context, info):
+        connection_type = info.return_type.graphene_type  # this is PostConnection
+        model = connection_type._meta.node._meta.model  # this is models.PostUnion
+        query = get_query(model, context)
+        discussion_id = context.matchdict['discussion_id']
+        query = query.filter(model.discussion_id == discussion_id
+            ).filter(model.hidden == False
+            ).filter(model.tombstone_condition()
+            ).filter(model.publication_state == models.PublicationStates.PUBLISHED)
+
+        # TODO filter posts for a specific idea
```

Comment:
Yeah we could have a idea_id argument to resolve_posts to only retrieve the posts from this idea. We don't need the ideas and posts for now. So we will discuss what we want in futures sprints. I disabled the resolve_ideas for now too. We can leave the resolve_posts as it is for now.